src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java

```diff
@@ -236,6 +244,180 @@ public void testFieldName() throws Exception {
+    InsertValidationResponse insertValidationResponse =
+        channel.insertRow(createStreamingIngestRow(row), UUID.randomUUID().toString());
+    assertThat(insertValidationResponse.getInsertErrors().size()).isEqualTo(1);
+    assertThat(insertValidationResponse.getInsertErrors().get(0).getExtraColNames())
+        .containsOnly(
+            "VALUE.list.element.key_value.value._k1",
+            "VALUE.list.element.key_value.value._k2",
+            "VALUE.list.element.key_value.value._k3",
+            "VALUE.list.element.key_value.value._k4",
+            "VALUE.list.element.key_value.value._k5",
+            "VALUE.list.element.key_value.value._k6",
+            "VALUE.list.element.key_value.value._k7",
+            "VALUE.list.element.key_value.value._k8",
+            "VALUE.list.element.key_value.value._k9",
+            "VALUE.list.element.key_value.value._k10",
+            "VALUE.list.element.key_value.value._k11",
+            "VALUE.list.element.key_value.value._k12");
+  }
+
+  @Test
+  public void testMissingFields() throws SQLException {
+    String tableName = createIcebergTable("object(k1 int not null, k2 int not null) not null");
+    SnowflakeStreamingIngestChannel channel =
+        openChannel(tableName, OpenChannelRequest.OnErrorOption.CONTINUE);
+
+    InsertValidationResponse insertValidationResponse =
+        channel.insertRow(createStreamingIngestRow(null), UUID.randomUUID().toString());
+    assertThat(insertValidationResponse.getInsertErrors().size()).isEqualTo(1);
+    assertThat(insertValidationResponse.getInsertErrors().get(0).getNullValueForNotNullColNames())
+        .containsOnly("VALUE");
+
+    insertValidationResponse =
+        channel.insertRow(
+            createStreamingIngestRow(new HashMap<String, Object>()), UUID.randomUUID().toString());
+    assertThat(insertValidationResponse.getInsertErrors().size()).isEqualTo(1);
+    assertThat(insertValidationResponse.getInsertErrors().get(0).getMissingNotNullColNames())
+        .containsOnly("VALUE.k1", "VALUE.k2");
+
+    insertValidationResponse =
+        channel.insertRow(
+            createStreamingIngestRow(
+                new HashMap<String, Object>() {
+                  {
+                    put("k1", null);
+                    put("k2", null);
+                  }
+                }),
+            UUID.randomUUID().toString());
+    assertThat(insertValidationResponse.getInsertErrors().size()).isEqualTo(1);
+    assertThat(insertValidationResponse.getInsertErrors().get(0).getNullValueForNotNullColNames())
+        .containsOnly("VALUE.k1", "VALUE.k2");
+  }
+
+  @Test
+  public void testMultipleErrors() throws Exception {
+    String tableName =
+        createIcebergTable(
+            "object(k1 int not null, k2 object(k3 int not null, k4 object(k5 int not null) not"
+                + " null) not null) not null");
+    SnowflakeStreamingIngestChannel channel =
+        openChannel(tableName, OpenChannelRequest.OnErrorOption.ABORT);
```

Comment:
looks like a pre-existing gap where someone using OnErrorOption.ABORT will not get the same set of information (insertErrors object) as OnErrorOption.CONTINUE ? (unrelated to your PR)

Reply:
For single row, errors on different OnErrorOption should be the same.